Makefile.am

```diff
@@ -25,7 +25,6 @@ libfastjsoninclude_HEADERS = \
 
 libfastjson_la_LDFLAGS = -version-info 3:0:0 -no-undefined @JSON_BSYMBOLIC_LDFLAGS@ \
 	-export-symbols-regex '^fjson_.*'
-libfastjson_la_CPPFLAGS = -Werror
```

Comment:
2016-04-07 16:09 GMT+02:00 Michael Biebl <notifications@github.com>:

> In Makefile.am
> <https://github.com/rsyslog/libfastjson/pull/47#discussion_r58877067>:
>
> > @@ -25,7 +25,6 @@ libfastjsoninclude_HEADERS = \
> >
> >  libfastjson_la_LDFLAGS = -version-info 3:0:0 -no-undefined @JSON_BSYMBOLIC_LDFLAGS@ \
> >  	-export-symbols-regex '^fjson_.*'
> > -libfastjson_la_CPPFLAGS = -Werror
>
> What exactly is the problem with passing -Werror via CFLAGS? Do you have
> any references?
>

The problem is that -Werror make autoconf AC_CHECK_FUNC checks think
features are not available, just because the way configure works. Check
here for a sample using strcmp:

https://gist.github.com/rgerhards/723c3777158fb882e15c9179c1fa76c8#file-config-log-L415

That said, I noticed that in the mean time I was able to remove all
AC_CHECK_FUNC, so it is not an issue at the moment.

> That said, a good middle ground could be something like [1].
> During development you could build with
> --enable-more-warnings=error.
>

That's probably a better method anyhow.

> The important part is, that hard-coding -Werror is bad and should be
> removed.
>
> [1]
> https://cgit.freedesktop.org/NetworkManager/NetworkManager/tree/m4/compiler_warnings.m4
>

read it and agree, do not question the need, but think about *how* to do
it. I need to say that I just recently added -Werror after I banged my head
for some hours in not getting travis up due to above-mentioned problem.

Rainer

> —
> You are receiving this because you were assigned.
> Reply to this email directly or view it on GitHub
> <https://github.com/rsyslog/libfastjson/pull/47/files/14be0062e7fe2f79d0428596c15b3e609d8d195a#r58877067>
>